code/src/terrat_migrations_ex/terrat_migrations_ex_150.ml

```diff
@@ -0,0 +1,327 @@
+module Sql = struct
+  let change_dirspaces_update_sql =
+    {|
+     with limited as (
+       select * from change_dirspaces where repo is null limit 10000
+     )
+     update change_dirspaces as cd
+     set repo = grm.core_id
+     from limited
+     inner join github_repositories_map as grm
+         on grm.repository_id = limited.repository
+     where (cd.repository, cd.base_sha, cd.sha, cd.path, cd.workspace) = (limited.repository, limited.base_sha, limited.sha, limited.path, limited.workspace)
+           and grm.repository_id = cd.repository
+     |}
+
+  let change_dirspaces_update = Pgsql_io.Typed_sql.(sql /^ change_dirspaces_update_sql)
+
+  let change_dirspaces_while_ =
```

Comment:
These `while` checks are here because a limitation of the pgsql io API is that we do not return the data around how many records were changed on an update/insert/delete.  We probably should but for right now working around it with a check to see if any records exist that have not been updated.